client/my-sites/site-settings/site-owner-transfer/site-owner-user-search.tsx

```diff
@@ -0,0 +1,103 @@
+import { Button, Gridicon } from '@automattic/components';
+import styled from '@emotion/styled';
+import { TextControl } from '@wordpress/components';
+import { useTranslate } from 'i18n-calypso';
+import { useState, FormEvent, useEffect } from 'react';
+import { useCheckSiteTransferEligibility } from './use-check-site-transfer-eligibility';
+
+const TextControlContainer = styled.div( {
+	marginBottom: '2em',
+} );
+
+const Error = styled.div( {
+	display: 'flex',
+	alignItems: 'center',
+	color: 'red',
+} );
+
+const ErrorText = styled.p( {
+	margin: 0,
+	marginLeft: '0.4em',
+	fontSize: '14px',
+} );
+
+const SiteOwnerTransferEligibility = ( {
+	selectedSiteId,
+	selectedSiteSlug,
+	siteOwner,
+	onNewUserOwnerSubmit,
+}: {
+	selectedSiteId: number;
+	selectedSiteSlug: string;
+	siteOwner: string;
+	onNewUserOwnerSubmit: ( user: string ) => void;
+} ) => {
+	const translate = useTranslate();
+	const [ tempSiteOwner, setTempSiteOwner ] = useState( siteOwner );
+	const [ siteTransferEligibilityError, setSiteTransferEligibilityError ] = useState( '' );
+
+	const { checkSiteTransferEligibility, isLoading: isCheckingSiteTransferEligibility } =
+		useCheckSiteTransferEligibility( selectedSiteId, {
+			onMutate: () => {
+				setSiteTransferEligibilityError( '' );
+			},
+			onError: () => {
+				setSiteTransferEligibilityError( translate( 'User not found' ) );
```

Comment:
The API error responses are fairly descriptive and helpful. Can we display them instead?